pystoned/CNLS.py

```diff
@@ -1,304 +1,224 @@
-                        if i == h:
-                            return Constraint.Skip
-                        return model.a[i] + sum(model.b[i, j] * arow[j] for j in model.j) >= model.a[h] + \
-                               sum(model.b[h, j] * arow[j] for j in model.j)
-
-                    model.qconvex = Constraint(model.i, model.h, rule=qconvex_rule, doc='convexity constraint')
-
-            if rts == "crs":
-
-                # Constraints
-                def qreg_rule(model, i):
-                    return log(y[i]) == log(model.f[i] + 1) + model.e[i]
-
-                model.qreg = Constraint(model.i, rule=qreg_rule, doc='log-transformed regression equation')
-
-                def qlog_rule(model, i):
-                    arow = x[i]
-                    return model.f[i] == sum(model.b[i, j] * arow[j] for j in model.j) - 1
-
-                model.qlog = Constraint(model.i, rule=qlog_rule, doc='regression function')
-
-                # production model
-                if fun == "prod":
-
-                    def qconcav_rule(model, i, h):
-                        arow = x[i]
-                        if i == h:
-                            return Constraint.Skip
-                        return sum(model.b[i, j] * arow[j] for j in model.j) <= \
-                               sum(model.b[h, j] * arow[j] for j in model.j)
-
-                    model.qconcav = Constraint(model.i, model.h, rule=qconcav_rule, doc='concavity constraint')
-
-                # cost model
-                if fun == "cost":
-
-                    def qconvex_rule(model, i, h):
-                        arow = x[i]
-                        if i == h:
-                            return Constraint.Skip
-                        return sum(model.b[i, j] * arow[j] for j in model.j) >= \
-                               sum(model.b[h, j] * arow[j] for j in model.j)
-
-                    model.qconvex = Constraint(model.i, model.h, rule=qconvex_rule, doc='convexity constraint')
-
-    return model
\ No newline at end of file
+# Import pyomp module
+from pyomo.environ import *
+from pyomo.core.expr.numvalue import NumericValue
+
+
+class CNLS:
+    def __init__(self, y, x, cet='addi', fun='prod', rts='vrs'):
+        # cet     = "addi" : Additive composite error term
+        #         = "mult" : Multiplicative composite error term
+        # fun     = "prod" : production frontier
+        #         = "cost" : cost frontier
+        # rts     = "vrs"  : variable returns to scale
+        #         = "crs"  : constant returns to scale
+
+        ## TODO(error/warning handling): Check the configuration of the model exist
+        self.x = x.tolist()
+        self.y = y.tolist()
+        self.cet = cet
+        self.fun = fun
+        self.rts = rts
+
+        if type(self.x[0]) != list:
+            self.x = [x.tolist()]
+
+        # Initialize the pyomo model
+        self.__model__ = ConcreteModel()
+        self.__model__.I = Set(initialize=range(len(self.y)))
```

Comment:
I would perfer to use "i" and "j" instead of "I" and "J", respectively.

Reply:
I have considered about this.
My thought while replacing the set name `i` to `I` is that:
* The variable name `i` represented index of DMU or we say an elements in a set.
* Originally, the set have the same name as the elements inside it.
* My consider is better not using same variable name to different units.
![image](https://user-images.githubusercontent.com/37362730/89163731-9b99ee80-d5a8-11ea-9bec-581e02b051c9.png)

Reply:
See next comment.